autometa/validation/datasets.py

```diff
@@ -0,0 +1,174 @@
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+    out_dirpath : str
+        directory path where the user wants to download the file(s)
+
+    Returns
+    -------
+    None
+        download is completed through gdown
+
+    """
+
+    key_list = [*targets_dict["file"]]
+    for key in key_list:
+        # retrieve values from targets_dict
+        file = targets_dict["file"][key]
+        folder = targets_dict["folder"][key]
+        file_id = targets_dict["file_id"][key]
+
+        # construct file id into a url to put into gdown
+        url = f"https://drive.google.com/uc?id={file_id}"
+        filename = f"{folder}_{file}"
+        out_fpath = os.path.join(
+            out_dirpath, filename
+        )  # this returns an error when relative file paths get used, eg "~/download"
+
+        # download the specified file with gdown
+        gdown.download(url, out_fpath)
+
+
+def main():
+    import argparse
+    import logging as logger
+
+    logger.basicConfig(
+        format="[%(asctime)s %(levelname)s] %(name)s: %(message)s",
+        datefmt="%m/%d/%Y %I:%M:%S %p",
+        level=logger.DEBUG,
+    )
+
+    parser = argparse.ArgumentParser(
+        prog="autometa-download-dataset",
+        description="Download a simulated community file from google drive to a specified directory",
+    )
+    parser.add_argument(
+        "--folder",
+        help="specify a simulated community size to download from (leave blank to download all)",
+        choices=[
+            "78.125Mbp",
+            "156.25Mbp",
+            "312.5Mbp",
+            "625Mbp",
+            "1250Mbp",
+            "2500Mbp",
+            "5000Mbp",
+            "10000Mbp",
+        ],
+    )
+    parser.add_argument(
+        "--file",
+        help="specify a file to download (or leave blank to download all)",
+        choices=[
+            "all",
+            "README.md",
+            "reference_assignments.tsv.gz",
+            "metagenome.fna.gz",
+            "master.tsv.gz",
+            "control_reads.tsv.gz",
+            "control_contigs.tsv.gz",
+            "unclustered_recruitment.tsv.gz",
+            "binning.tsv.gz",
+            "taxonomy.tsv.gz",
+            "lengths.tsv.gz",
+            "coverages.tsv.gz",
+            "gc_content.tsv.gz",
+            "kmers.embedded.tsv.gz",
+            "kmers.tsv.gz",
+            "markers.tsv.gz",
+            "Bacteria.fna.gz",
+            "orfs.faa.gz",
+            "metagenome.filtered.fna.gz",
+            "hmmscan.tsv.gz",
+            "forward_reads.fastq.gz",
+            "reverse_reads.fastq.gz",
+        ],
+    )
+    parser.add_argument(
+        "--out_dirpath",
+        help="specify the directory to download the file(s)",
+        required=True,
+    )
+    args = parser.parse_args()
+
+    download(fetchIDs(args.folder, args.file), args.out_dirpath)
```

Comment:
I have fallen into a similar trap here in the past. Consider splitting up some of your main logic. Ultimately it should be readable (_almost like pseudocode_) in describing its functionality. You've got some of the logic implemented into your functions, but now that they're nested, it has become a bit more difficult to digest. See below for an example:

Example bash call:

```bash
autometa-datasets \
    --community 78 \
    --file metagenome,reference_assignments,binning \
    --output $HOME/78Mbp.metagenome.fna.gz,$HOME/78Mbp.assignments.tsv.gz,$HOME/78Mbp.binning.tsv.gz
```

NOTE: It may be easiest to consider only allowing the user to specify one community. This would play nicely with nextflow and simplify your task. Another thing to notice from the above example is the number of arguments supplied to `--file` and `--output` are the same. They correspond to each other such that we can look them up in our python code.

e.g.

```python
def main():
    ############
    args = parser.parse_args()
    dataset = args.community
    filetypes = args.file
    filepaths = args.output
    ############
    # NOTE: You will need to look up some argparse docs for the above to properly work (see `nargs`)
    for filetype, filepath in zip(filetypes, filepaths):
        # The following behavior is more intuitive. It almost reads out as a sentence
        download(dataset=dataset, file=filetype, out=filepath)
        # "download" the "dataset" of the "filetype" to "filepath"
```

You could similarly use your `fetchIDs(...)` function, but I would break this up in the `main` logic.


```python
file_ids = fetchIDs(...)
for file_id in file_ids:
    # Notice we have no method here to specify
    # the output file path which is behavior that 
    # would be convenient for the end-user.
    download(..., file_id, ...)
```

Reply:
Now you can select one folder, multiple files, and the main logic is cleaner. Thanks for the suggestions!